### PR TITLE
Add typing for core-js string.prototype.padStart and padEnd

### PR DESCRIPTION
**Pull request by stewx**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/zloirock/core-js
- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/core-js/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for core-js 0.9
+// Type definitions for core-js 0.9.1
```
Comment (andy-ms) [inline types/core-js/index.d.ts]: We don't parse a patch version here, since the published patch version is automatically incremented on every change.

Comment (stewx) [reply types/core-js/index.d.ts]: What am I supposed to do here?

Comment (stewx) [reply types/core-js/index.d.ts]: I've updated it with a minor version bump

File: types/core-js/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for core-js 0.9
+// Type definitions for core-js 0.10
```
Comment (andy-ms) [inline types/core-js/index.d.ts]: It looks like `core-js 0.10` was never released and it went straight from `0.9` to `1.0`. You don't *have* to update the header if the library version didn't change.

Comment (stewx) [reply types/core-js/index.d.ts]: The core-js project is up past 2.0 now. I thought this was the version number for the typing. Should I revert this? I am not claiming that my changes bring this typing in line with a particular version of core-js. I am just adding a couple missing function definitions.

Comment (andy-ms) [reply types/core-js/index.d.ts]: Yeah, change it back then.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**